internal/controllers/networking/loadbalancer_controller.go

```diff
@@ -71,6 +75,17 @@ func (r *LoadBalancerReconciler) delete(ctx context.Context, log logr.Logger, lo
 func (r *LoadBalancerReconciler) reconcile(ctx context.Context, log logr.Logger, loadBalancer *networkingv1alpha1.LoadBalancer) (ctrl.Result, error) {
 	log.V(1).Info("Reconcile")
 
+	if loadBalancer.Spec.Type == networkingv1alpha1.LoadBalancerTypeInternal {
+		ips, err := r.applyInternalIPs(ctx, log, loadBalancer)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error getting / applying internal ip: %w", err)
+		}
+		if err := r.updateStatus(ctx, log, loadBalancer, ips); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error patching load balancer status")
+		}
+		log.V(1).Info("Patched load balancer status with internal ip")
```

Comment:
Could you please move the `updateStatus` call to the bottom of the `reconcile` method? To 'save' the IPs, do like following pseudo code

```go
var ips []commonv1alpha1.IP
if loadBalancer.Spec.Type == networkingv1alpha1.LoadBalancerTypeInternal {
	var err error
	ips, err = r.applyInternalIPs(ctx, log, loadBalancer)
	if err != nil {
		return ctrl.Result{}, fmt.Errorf("error getting / applying internal ips: %w", err)
	}
}
...
if err := r.updateStatus(ctx, log, loadBalancer, ips); err != nil {
	return ctrl.Result{}, fmt.Errorf("error patching load balancer status: %w")
}
log.V(1).Info("Updated load balancer status")

Reply:
If `loadBalancer.Spec.NetworkInterfaceSelector` is not provided, `loadBalancer.Status` wont get updated.